src/utils.js

```diff
@@ -58,3 +53,73 @@ function getHashParams (link) {
   hashParamsObj.key = Buffer.from(hashParamsObj.key, 'base64')
   return hashParamsObj
 }
+
+/**
+   * Get a value
+   * @param {Object} db - The hyperDB instance
+   * @param {CryptoKey} enckey - The enc/dec AES key
+   * @param {string} key - The key name
+   * @returns {Promise}
+   */
+const get = async (db, encKey, key) => {
+  if (!db) throw generateError(ERRORS.NODB)
+  if (!encKey) throw generateError(ERRORS.NOENCRYPTIONKEY)
+  const node = await db.getAsync(key)
+  if (!node) return null
+  const dec = await decrypt(encKey, node.value)
+  return dec
+}
+
+/**
+   * Put a new value in the db
+   * @param {Object} db - The hyperDB instance
+   * @param {CryptoKey} enckey - The enc/dec AES key
+   * @param {string} key - The key name
+   * @param {any} value - The value to insert
+   * @returns {Promise}
+   */
+const put = async (db, encKey, key, value) => {
+  if (!db) throw generateError(ERRORS.NODB)
+  if (!encKey) throw generateError(ERRORS.NOENCRYPTIONKEY)
+  const enc = await encrypt(encKey, value)
+  return db.putAsync(key, enc)
+}
+
+/**
+   * List all keys and values
+   * @param {Object} db - The hyperDB instance
+   * @param {CryptoKey} enckey - The enc/dec AES key
+   * @param {string} prefix - Prefix
+   * @returns {Promise}
+   */
+const list = async (db, encKey, prefix) => {
+  if (!db) throw generateError(ERRORS.NODB)
+  if (!encKey) throw generateError(ERRORS.NOENCRYPTIONKEY)
+
+  const list = await db.listAsync(prefix)
+  if (list.length === 1 && list[0].key === '' && list[0].value === null) {
+    return {}
+  }
+
+  const decList = await Promise.all(list.map(async (elt) => ({
+    key: elt.key,
+    value: await decrypt(encKey, elt.value)
+  })))
+
+  const reformattedDic = decList.reduce((dic, e) => {
+    const el = Array.isArray(e) ? e[0] : e
```

Comment:
In case we have multiple nodes right ?

Reply:
Good remark, because we force first node true, this check is no longer required.